src/main/kotlin/no/nav/familie/ks/sak/kjerne/overgangsordning/OvergangsordningAndelValidator.kt

```diff
@@ -0,0 +1,156 @@
+package no.nav.familie.ks.sak.kjerne.overgangsordning
+
+import no.nav.familie.ks.sak.common.exception.FunksjonellFeil
+import no.nav.familie.ks.sak.common.tidslinje.filtrerIkkeNull
+import no.nav.familie.ks.sak.common.tidslinje.tilTidslinje
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.kombinerMed
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.tilPerioder
+import no.nav.familie.ks.sak.common.util.tilDagMånedÅrKort
+import no.nav.familie.ks.sak.common.util.tilMånedÅrKort
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Resultat
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårResultat
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.tilTidslinje
+import no.nav.familie.ks.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.beregning.domene.ordinæreAndeler
+import no.nav.familie.ks.sak.kjerne.beregning.domene.overgangsordningAndeler
+import no.nav.familie.ks.sak.kjerne.overgangsordning.domene.OvergangsordningAndel
+import no.nav.familie.ks.sak.kjerne.overgangsordning.domene.UtfyltOvergangsordningAndel
+import no.nav.familie.ks.sak.kjerne.overgangsordning.domene.tilPerioder
+import no.nav.familie.ks.sak.kjerne.personident.Aktør
+import java.time.YearMonth
+
+object OvergangsordningAndelValidator {
+    fun validerAndelerErIPeriodenBarnetEr20Til23Måneder(
+        andeler: List<UtfyltOvergangsordningAndel>,
+    ) {
+        andeler
+            .groupBy { it.person }
+            .forEach { (person, andelerForPerson) ->
+                val tidligsteGyldigeFom = beregnGyldigFom(person)
+                val senesteGyldigeTom = beregnGyldigTom(person)
+                andelerForPerson.forEach {
+                    if (it.fom.isBefore(tidligsteGyldigeFom) || it.tom.isAfter(senesteGyldigeTom)) {
+                        throw FunksjonellFeil(
+                            melding = "Perioden som blir forsøkt lagt til er utenfor gyldig periode for person.",
+                            frontendFeilmelding =
+                                "Perioden for overgangsordning du forsøker å legge til (${it.fom} - ${it.tom}) er utenfor gyldig periode for barn født ${person.fødselsdato}. " +
+                                    "Gyldig periode er fra og med 20 til og med 23 måneder etter fødselsdato ($tidligsteGyldigeFom - $senesteGyldigeTom).",
+                        )
+                    }
+                }
+            }
+    }
+
+    fun validerAtAlleOpprettedeOvergangsordningAndelerErUtfylt(andeler: List<OvergangsordningAndel>) {
+        runCatching {
+            andeler.forEach { it.validerAlleFelterUtfylt() }
```

Comment:
Siden denne kjører validerAlleFelterUtfylt(), som også kan kaste feil fordi feltene er gyldige, så tror jeg kanskje at det kan bli litt misvisende for SB.

Gitt at validerAlleFelterUtfylt kaster feilen med melding "Du kan ikke sette en t.o.m. dato som er før f.o.m. dato"", så vil denne kaste feilen "Du har opprettet en eller flere perioder for overgangsordning som er ufullstendig utfylt.". Tror jeg 🤔

Reply:
Har splittet opp slik at den enten kaster den lange feilmeldingen, dersom noen felter ikke er utfylt, eller en spesifikk melding dersom tom > fom eller antallTimer < 0. Ettersom det er validering i frontend og i servicen som oppdaterer hver andel, bør valideringene som sjekker om feltene er gyldige aldri treffes